packages/shared/src/hooks/AGENTS.md

```diff
@@ -0,0 +1,207 @@
+# Hooks
+
+Custom React hooks for reusable logic across the application.
+
+## Organization
+
+Hooks are organized by domain:
+
+| Directory | Purpose |
+|-----------|---------|
+| `auth/` | Authentication hooks |
+| `bookmark/` | Bookmark operations |
+| `feed/` | Feed data and interactions |
+| `post/` | Post operations |
+| `comments/` | Comment operations |
+| `notifications/` | Notification settings |
+| `search/` | Search functionality |
+| `source/` | Source/squad operations |
+| `squads/` | Squad-specific hooks |
+| `vote/` | Voting operations |
+| `log/` | Analytics logging |
+| Root level | General-purpose hooks |
+
+## Naming Convention
+
+```
+use[Domain][Action].ts
+
+Examples:
+- useBookmarkPost.ts
+- useFeedSettings.ts
+- usePostById.ts
+- useShareComment.ts
+```
+
+## TanStack Query Patterns
+
+Most data-fetching hooks use TanStack Query v5:
```

Comment:
I would actually prefer our new typed queries and mutations logic https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/1556185091/Data+fetching+mutations+and+optimistic+updates+with+typed+queries cc: @rebelchris I think you also agree its a good direction, especially for cache invalidations?